graphql_api/types/bundle_analysis/base.py

```diff
@@ -135,6 +145,65 @@ def resolve_assets(
     return list(bundle_report.assets())
 
 
+@bundle_report_bindable.field("assetsPaginated")
+def resolve_assets_paginated(
+    bundle_report: BundleReport,
+    info: GraphQLResolveInfo,
+    ordering: AssetOrdering = AssetOrdering.SIZE,
+    ordering_direction: OrderingDirection = OrderingDirection.DESC,
+    first: Optional[int] = None,
+    after: Optional[str] = None,
+    last: Optional[int] = None,
+    before: Optional[str] = None,
+) -> Dict[str, object]:
+    # All filtered assets before pagination
+    assets = list(
+        bundle_report.assets(
+            ordering=ordering.value,
+            ordering_desc=ordering_direction.value == OrderingDirection.DESC.value,
+        )
+    )
+
+    total_count, has_next_page, has_previous_page = len(assets), False, False
+    start_cursor, end_cursor = None, None
+
+    # Apply cursors to edges
+    if after is not None:
+        after_edge = _find_index_by_cursor(assets, after)
+        if after_edge > -1:
+            assets = assets[after_edge + 1 :]
+
+    if before is not None:
+        before_edge = _find_index_by_cursor(assets, before)
+        if before_edge > -1:
+            assets = assets[:before_edge]
+
+    # Slice edges by return size
+    if first is not None and first >= 0:
+        if len(assets) > first:
```

Comment:
nit: we can use total_count var instead of len(assets) for all these references to len(assets) right?

Reply:
total_count might not equal to len(assets) at this point of the code path. assets could have gotten sliced by the after/before if statements. total_count is for the grand total of results after filtreing.

Reply:
ahhh okay yeah I totally missed that previously, makes sense